src/CattyTests/DataModel/UserVariableTests.swift

```diff
@@ -70,4 +73,91 @@ final class UserVariableTests: XCTestCase {
         XCTAssertFalse(userVariableA === userVariableCopy)
         XCTAssertTrue(userVariableB === userVariableCopy)
     }
+
+    func testIsEqualToUserVariableForDirectAssignment() {
+        let userVariableA = UserVariable()
+        userVariableA.name = "userVar"
+        userVariableA.value = self.arrayWithSameElement
+
+        var userVariableB = userVariableA
+        XCTAssertTrue(userVariableA.isEqual(to: userVariableB))
+        XCTAssertEqual(userVariableB.value as! NSMutableArray, self.arrayWithSameElement)
+
+        userVariableA.value = self.arrayWithDifferentElement
+        XCTAssertTrue(userVariableA.isEqual(to: userVariableB))
+        XCTAssertEqual(userVariableB.value as! NSMutableArray, self.arrayWithDifferentElement)
+
+        let userVariableC = UserVariable()
+        userVariableC.name = "userVarC"
+
+        userVariableB = userVariableC
+        XCTAssertFalse(userVariableA.isEqual(to: userVariableB))
+    }
+
+    func testIsEqualToUserVariableForInitWithNameAndIsList() {
+        let userVariableA = UserVariable()
+        userVariableA.name = "userVar"
+        userVariableA.value = self.arrayWithSameElement
+
+        var userVariableB = UserVariable(name: userVariableA.name, isList: userVariableA.isList)
+        XCTAssertFalse(userVariableB!.isEqual(to: userVariableA))
+        XCTAssertNil(userVariableB?.value)
+
+        userVariableB?.value = userVariableA.value
+        XCTAssertTrue(userVariableB!.isEqual(to: userVariableA))
+        XCTAssertEqual(userVariableB?.value as! NSMutableArray, self.arrayWithSameElement)
+
+        userVariableA.value = self.arrayWithDifferentElement
+        userVariableB?.value = userVariableA.value
+        XCTAssertTrue(userVariableA.isEqual(to: userVariableB))
+        XCTAssertEqual(userVariableB?.value as! NSMutableArray, self.arrayWithDifferentElement)
+
+        userVariableB = UserVariable(name: userVariableA.name, isList: !userVariableA.isList)
+        XCTAssertFalse(userVariableB!.isEqual(to: userVariableA))
+    }
+
+    func testIsEqualToUserVariableForInitWithUserVariable() {
+        let userVariableA = UserVariable()
+        userVariableA.name = "userVar"
+        userVariableA.value = self.arrayWithSameElement
+
+        var userVariableB = UserVariable(variable: userVariableA)
+        XCTAssertFalse(userVariableB!.isEqual(to: userVariableA))
+        XCTAssertNil(userVariableB?.value)
+
+        userVariableB?.value = userVariableA.value
```

Comment:
This assignment is not making a deep copy of your elements, hence the test is not testing what it should. Please directly assign the array to this variable to get a new instance (i.e., `NSMutableArray(array: [1, 2])`).

Reply:
This applies for some other methods as well.